### ISSUE TITLE
Redirect path to domain.com/web is not working (Docker Swarm Mode 1.13)

### ISSUE THREAD

**Issue by pascalandy** [containous/traefik#1123]
### What version of Traefik are you using (`traefik version`)?
traefik:v1.1.2-alpine

### What is your environment & configuration (arguments, toml...)?
https://gist.github.com/pascalandy/a28958d8794a6c142a11accb7623b817

### Intro
Hey folks!

As seen in the [concept section](https://docs.traefik.io/basics/), I want to create a path as:
> path domain.com/web will point the microservice web in your private network

*I can successfully redirect the path over `api.domain.com`. Here, I am strugling to redirect the path  over `domain.com/web`.*

### What did you do?

1) Here is how I create my **nginx** container:

```
CTN_NGX_TEST="nginxtest"
WWW_SRC="/to-mount/nginx-static/starter"
WWW_DST="/usr/share/nginx/html"

docker service create \
--name "$CTN_NGX_TEST" \
--network "$NTW_FRONT" \
--mount	type=bind,src="$WWW_SRC",target="$WWW_DST" \
--label traefik.enable="true" \
--label traefik.frontend.rule="Host:example.com;PathPrefixStrip:/nginxtest" \
--label traefik.frontend.entryPoints="http" \
--label traefik.docker.network="$NTW_FRONT" \
--label traefik.backend="$CTN_NGX_TEST" \
--label traefik.backend.loadbalancer.sticky="true" \
--label traefik.weight="10" \
--label traefik.frontend.priority="10" \
--label traefik.backend.loadbalancer.method="drr" \
--label traefik.port="80" \
nginx
```

2) Here is how I created my **Traefik** container.
http://pascalandy.com/blog/gist-2017-02-04_12h18

### The Issue
[See the screenshot](http://blog.pascalandy.com/gist-2017-02-04_12h18/). From this page, if I click on the link about, the browser points to to: `example.com/about/about-1.html`. Can you see the issue??

### What did you expect to see?
`example.com/nginxtest/about/about-1.html`

### What did you see instead?
`example.com/about/about-1.html`

**Interesting**: If I manually enter `example.com/nginxtest/index.html` in my address bar, I can surf normally within the website! 

So what is wrong ??

Thank you in advance :)

**Comment by pascalandy** [comment]
My @emilevauge ,
After searching/trying for hours, I found that manu issues are related.

The main issue is about the “/“ that mess with the routings. I believe [issue 563](https://github.com/containous/traefik/issues/563) nails it perfectly.

For API calls, this is not an issue. Coders can hardcode “/“ in their path. 

But for public sites, the fact that example.com/demo is not working is a no go for production. Grrhh.


### Related issue:
Hope it can help to see how serious this is. I also saw few tickets closed [1](https://github.com/containous/traefik/issues/500#issuecomment-231387556), [2](https://github.com/containous/traefik/issues/337#issuecomment-215495621), about the same topic.

https://github.com/containous/traefik/issues/563
https://github.com/containous/traefik/issues/374
https://github.com/containous/traefik/issues/850
https://github.com/containous/traefik/issues/920
https://github.com/containous/traefik/issues/1037
https://github.com/containous/traefik/issues/1011
https://github.com/containous/traefik/issues/1123

Cheers!

**Comment by pascalandy** [comment]
Hi @emilevauge ,
After searching/trying for hours, I found that many issues are related.

The main issue is about the “/“ that mess with the routings. I believe issue 563 nails it perfectly.

For API calls, this is not an issue. Coders can hardcode “/“ in their path.

But for public sites, the fact that example.com/demo is not working is a no go for production. Grrhh.

Related issue:

Hope it can help to see how serious this is. I also saw few tickets closed 1, 2, about the same topic.

#563
#374
#850
#920
#1037
#1011
#1123

Cheers!

**Comment by emilevauge** [comment]
@pascalandy For the record: https://github.com/containous/traefik/issues/563#issuecomment-257604474

**Comment by deitch** [comment]
I am banging my head on the wall with this one. The problem seems to be that it would require rewriting the body of the response. It can be done (I think nginx does it with `sub_filter`), but is non-trivial and is likely to be trouble. And what do you do with dynamically created tags?

Truth is, well-written Web sites should almost never be tied to absolute root of `/images/abc.png` or whatever, but the real world is, well, lots of apps that do.

Sigh.

**Comment by deitch** [comment]
Actually, everything Webpack ever does exactly that. Now I am dealing with a packaged container that always serves out its assets from `/app/dist/`, which means path-based routing always fails.

**Comment by pascalandy** [comment]
In the first scenario, the website appears offline. So I need to hack my check routine and manually add the '/' add the and of the domain to get a valide curl test.

### 1) Without the '/' 

root@N02-par1-dev-e:~/deploy-setup# curl -Is --head mysite.com/u/zombie-crafty/news
HTTP/1.1 301 Moved Permanently
Date: Wed, 01 Mar 2017 16:35:13 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Set-Cookie: __cfduid=d3f5ae298c250aedc8bf8d9c7870c2b5b1488386113; expires=Thu, 01-Mar-18 16:35:13 GMT; path=/; domain=.cloudwait.com; HttpOnly
Location: /u/zombie-crafty/news/
Set-Cookie: _TRAEFIK_BACKEND=http://111.22.0.8:2368
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Powered-By: Express
Server: cloudflare-nginx
CF-RAY: 338d873b06351031-CDG

### 2) With the '/' 

root@N02-par1-dev-e:~/deploy-setup# curl -Is --head mysite.com/u/zombie-crafty/news/
HTTP/1.1 200 OK
Date: Wed, 01 Mar 2017 16:35:24 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Set-Cookie: __cfduid=db5eba87a69b720bde5fcc1cb1fb8eded1488386124; expires=Thu, 01-Mar-18 16:35:24 GMT; path=/; domain=.cloudwait.com; HttpOnly
Cache-Control: public, max-age=0
Set-Cookie: _TRAEFIK_BACKEND=http://111.22.0.8:2368
Vary: Accept-Encoding
X-Powered-By: Express
Server: cloudflare-nginx
CF-RAY: 338d877e229368b4-CDG

**Comment by ldez** [comment]
Closed by #1638, in the version 1.3.0-rc3.

**Comment by Fank** [comment]
Same issue here:
1. traefik.frontend.rule: Host:foo.bar,bin.baz
2. traefik.frontend.rule: Host:foo.bar:PathPrefix:/sidebar
3. traefik.frontend.rule: Host:bin.baz:PathPrefix:/ws
4. traefik.frontend.rule: Host:bin.baz:PathPrefix:/v2

Every request will be forwarded to 1.
